### PR TITLE
set parent_id when overwriting metadata.parent_uid

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
This is a followup to https://github.com/mitodl/ocw-to-hugo/issues/482, toward #1041 

#### What's this PR do?
When using `overwrite_ocw_course_content` to import `metadata.parent_uid`, also set `parent_id`.

#### How should this be manually tested?

1. Have `6-004-computation-structures-spring-2017` locally
2. Inspect WebsiteContent object `c675efdb-a53d-c4ec-1d14-46b3d44fda1e` in the admin panel
3. Observe that it has `metadata.parent_title` but no `metadata.parent_uid` or `parent` 
4. Run `docker-compose run --rm web python manage.py overwrite_ocw_course_content --bucket ocw-to-hugo-output-qa  --content-field="metadata.parent_uid" --filter 6-004-computation-structures-spring-2017`
5. Inspect the WebsiteContent object again and observe that it now has *both* `metadata.parent_uid` and `parent`. 

If you do not have `6-004-computation-structures-spring-2017` locally, do the above with any object that has metadata `parent_title` but not `parent_uid`. There are lots.

### CODE REVIEW

File: ocw_import/api.py [hunk 1]
```diff
@@ -149,33 +143,45 @@ def update_ocw2hugo_content(
...
+    if is_metadata_field:
+        set_dict_field(
+            content_file.metadata,
+            update_field,
+            get_dict_field(
+                content_data.get("metadata", {}), update_field
+            ),
+        )
+        if update_field == "parent_uid" and content_data["parent"]:
+            content_file.parent_id = content_data["parent"].id
```
Comment (ChristopherChudzicki) [inline ocw_import/api.py:176]: This is the real change. I moved this all to a separate function partly for sanity and partly to appease the linting gods.

**Review by mbertrand — CHANGES_REQUESTED** [review]
When I initially imported `mas-962-special-topics-new-textiles-spring-2010` on master branch with the `import_ocw_course_sites` command, the resource `b4c023d4-6c9a-bffc-eb02-09357443a4ef` already had a populated `parent_uid` and `parent`:

Screen Shot 2022-03-16 at 9 51 04 AM

So I'm not sure how this resource got to its current state on RC/prod.  But with this PR, `overwrite_ocw_course_content` does populate `parent` if it isn't already, unlike before.

Unless this needs to go out ASAP, might be good to add a unit test for `update_content_from_s3_data ` to verify that `parent` is populated if `update_field` = "parent_uid"

**Comment by pdpinch** [comment]
Assigning this back to Chris since it's "waiting on author"

**Comment by mbertrand** [comment]
@pdpinch typically it remains assigned to the reviewer, just the label switches back and forth between "needs review" and "waiting on author"

File: ocw_import/api_test.py [hunk 2]
```diff
@@ -545,3 +549,84 @@ def test_update_ocw2hugo_course_content(
             ).count()
             == (1 if create_new_content else 0)
         )
+
+
+class TestUpdateContentFromS3Data:
```
Comment (ChristopherChudzicki) [inline ocw_import/api_test.py:554]: @mbertrand Let me know if you have any suggestions about this test. 

I wanted to be fairly explicit with assertions (e.g., assert against literals not computed expected dictionaries) but also wanted to not copy paste bunch of stuff between tests, hence the shared setup is in this class.

The `update_ocw2hugo_content` function which is more directly used by the management commands still has no tests, but at least this is something :/

**Review by mbertrand — APPROVED** [review]
Looks good to me including tests 👍

**Pull request merged**